tests/test.cpp

```diff
@@ -213,10 +214,58 @@ TEST_CASE( "rename-lock", "[basic]" ) {
 }
 
 
+TEST_CASE( "rename-on-top", "[basic]" ) {
+    auto h = std::make_shared<HiLok>('/', HiFlags::RECURSIVE);
+    auto l1 = h->write(h, "a/b/c/d");
+    auto l2 = h->write(h, "a/b/c");
+    h->rename("a/b/c/d", "a/b/c", false);
+   
+    // a/b/c write locked
+    CHECK(thread_check_write_locked(h, "a/b/c"));
+
+    l1->release();
+    l2->release();
+
+    CHECK(h->size() == 0);
+}
+
+TEST_CASE( "loose-read-unlock", "[basic]" ) {
+    auto h = std::make_shared<HiLok>('/', HiFlags::LOOSE_READ_UNLOCK | HiFlags::RECURSIVE);
+    auto l1 = h->read(h, "a/b/c");
+    auto fut = std::async(std::launch::async, &HiHandle::release, l1);
+    fut.get();
+}
+
+TEST_CASE( "loose-write-unlock", "[basic]" ) {
+    auto h = std::make_shared<HiLok>('/', HiFlags::LOOSE_WRITE_UNLOCK | HiFlags::RECURSIVE);
+    auto l1 = h->write(h, "a");
+    auto fut = std::async(std::launch::async, &HiHandle::release, l1);
+    fut.get();
+}
+
+TEST_CASE( "rename-read-deep", "[basic]" ) {
```

Comment:
Really dumb case that I feel like we're inevitably going to run into at some point: renaming a key to itself. Can probably just raise, but still suspect we should handle.

Reply:
it does nothing, but i will add a test